### PR TITLE
ui: Add peer searching and sorting

### PR DESCRIPTION
**Pull request by johncowen**
### Description
Adds initial name search and sort only, more to come here in terms of searching, sorting and filtering options.

~P.S. Tests here are currently failing for unrelated reasons, that is being worked on separately and we can rebase here once thats ready (see https://github.com/hashicorp/consul/pull/13633).~ 13633 was merged and rebased here, tests are green now.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

/cc @LevelbossMike

### CODE REVIEW

File: ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          <dt>{{key}}</dt>
+          <dd>{{value}}</dd>
+        </dl>
+      </search.RemoveFilter>
+{{/let}}
+
+    </:status>
+    <:search as |search|>
+      <search.Search
+        @onsearch={{action @onsearch}}
```
Comment (LevelbossMike) [inline ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs:39]: no reason to prefix these things with `action` anymore.

File: ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        @value={{@search}}
+        @placeholder={{t "common.search.search"}}
+      >
+        <search.Select
+          class="type-search-properties"
+          @position="right"
+          @onchange={{action @filter.searchproperty.change}}
```
Comment (LevelbossMike) [inline ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs:46]: no action required

File: ui/packages/consul-ui/app/templates/dc/peers/index.hbs [hunk 3]
```diff
@@ -3,59 +3,111 @@
...
+      value=(if (not-eq searchproperty undefined)
+        (split searchproperty ',')
+        searchProperties
+      )
+      change=(action (mut searchproperty) value="target.selectedItems")
+      default=searchProperties
+    )
+  )
+
+  peers
```
Comment (LevelbossMike) [inline ui/packages/consul-ui/app/templates/dc/peers/index.hbs:24]: `this.peers` or `@model.peers`

File: ui/packages/consul-ui/app/templates/dc/peers/index.hbs [hunk 4]
```diff
@@ -3,59 +3,111 @@
...
       <h1>
         <route.Title @title="Peers" />
       </h1>
     </BlockSlot>
     <BlockSlot @name="toolbar">
-      <Peerings::Search @search={{this.filter}} @onSearch={{this.handleSearchChanged}} />
+      {{#if (gt items.length 0)}}
+        <Consul::Peer::SearchBar
+          @search={{search}}
+          @onsearch={{action (mut search) value="target.value"}}
```
Comment (LevelbossMike) [inline ui/packages/consul-ui/app/templates/dc/peers/index.hbs:37]: using `pick` and `fn` here would be more idiomatic here in Octane but it's ok to change that in one go for all search related views.

File: ui/packages/consul-ui/app/templates/dc/peers/index.hbs [hunk 5]
```diff
@@ -3,59 +3,111 @@
...
+          </ListCollection>
+          </collection.Collection>
+          <collection.Empty>
+          {{!-- TODO: do we need to check permissions here or will we receive an error automatically? --}}
+          <EmptyState
+            @login={{route.model.app.login.open}}
+          >
+            <BlockSlot @name="header">
+              <h2>
+                {{#if (gt items.length 0)}}
```
Comment (LevelbossMike) [inline ui/packages/consul-ui/app/templates/dc/peers/index.hbs:76]: is this the other way around? i.e when we have more than one item we want to show the Welcome message?

**Review by LevelbossMike — CHANGES_REQUESTED** [review]

File: ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs (hunk 2 above)
Comment (johncowen) [reply ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs:46]: This comment was quite confusing 😆 "no action required" 😛

**Review by evrowe — APPROVED** [review]

**Pull request merged**